sqlmesh/core/config/connection.py

```diff
@@ -948,6 +937,15 @@ def _static_connection_kwargs(self) -> t.Dict[str, t.Any]:
         else:
             raise ConfigError("Invalid BigQuery Connection Method")
 
+        if self.impersonated_service_account:
+            from google.auth import impersonated_credentials
```

Comment:
Might as well move this up with the other google imports. Feels strange having it be imported when needed when the others are imported at the top of the method